pkg/tfgen/generate.go

```diff
@@ -1943,6 +1949,44 @@ func emitFile(fs afero.Fs, relPath string, contents []byte) error {
 	return err
 }
 
+func getUniqueLeafDocsDescriptions(arguments map[docsPath]*argumentDocs, path docsPath) string {
+	leaf := path.leaf()
+
+	uniqueLeaves := getUniqueLeafPaths(arguments)
+	if _, exist := uniqueLeaves[leaf]; exist {
+		for argKey, argDoc := range arguments {
+			if argKey.leaf() == leaf {
+				return argDoc.description
+			}
+		}
+	}
```

Comment:
This feels a little funny to me. We want to check `path.leaf()` is unique in `arguments`. The way we do that here is to:
1. Build a list of unique arguments.
2. Check if `path.leaf()` is in that list.
3. If it is, iterate through `arguments` again to find the description.

I think we could do this in a faster and simpler way:
```suggestion
	leaf := path.leaf()
	var foundArgumentDocs *argumentDocs
	for argKey, argDoc := range arguments {
		if argKey.leaf() != leaf {
			continue
		}
		if foundArgumentDocs != nil {
			// path.leaf() is not a unique leaf in arguments
			return ""
		}
		foundArgumentDocs = argDoc
	}
	// Could not find an argument suffixed by path.leaf()
	if foundArgumentDocs == nil {
		return ""
	}
	return foundArgumentDocs.description
```

We then wouldn't need `getUniqueLeafPaths` at all.